### DISCUSSION TITLE
Get Genre of Song

### DISCUSSION THREAD

**Discussion by ManuelBanza** [johnwmillr/LyricsGenius#179] [Q&A] [answered]
Hi,

Is there anyway to get the genre of a song? I look in the documentation and all I could find related to genre was charts.

Thanks

**Comment by allerter** [comment, marked as answer]
Hi. Genius doesn't provide genres (or _tags_ as Genius likes to call them) through their official API. However, you can get them through the public/user API. Here's how you can do it:
```python
from lyricsgenius import PublicAPI
song_id = 1  # the song you want tags for
public_api = PublicAPI()
song = public_api.song(song_id)['song']
print(song['tags'])
```
**Note**: You must be using v3.0 for this to work.
**Warning**: Personally, I haven't had any problems so far using the user/web API, but here's what one of the staff at [Genius](https://genius.com/discussions/267781-Search-api-results) said about it:
> Using the web API for non-browser based requests is not supported, and people found to be misusing it to gain information not provided by our public API [developers API] may be banned at any time with no warning."

**Reply by ManuelBanza** [reply]
Sorry for the late reply. Thank you very much!
Any idea how can I make this to a list containing just the name tag?

I have a list just with IDs and I would like to get a new list with the tag for each one of them, I tried this but it gets an error list index out of range

`` from lyricsgenius import PublicAPI
    public_api = PublicAPI()
    
    list_of_ids = df['id'].to_list()
    
    tags = []
    for song_id in list_of_ids:
        song = public_api.song(song_id)['song']['tags'][1]['name']
        tags.append(song)
``

**Reply by allerter** [reply]
You can use list comprehension:
```python
song = public_api.song(song_id)['song']
song_tags = [tag['name'] for tag in song['tags']]
```

**Comment by johnwmillr** [comment]
@Allerter, it is for this reason that I'd like to minimize use of the user/web API going forward, prioritizing the token-authorized API whenever possible.

**Comment by allerter** [comment]
@johnwmillr I agree with you, but it's apparent that Genius has abandoned the developers API for years now and they have no intention of moving the functionality from the web API to the developers API. They abandoned it so early that even for something as simple as getting the genres of a song we have to resort to the web API.
